checkstyle-tester/README.md

```diff
@@ -121,6 +121,20 @@ With all that done, you can now call groovy. **baseBranch** will be your PR bran
 groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch i111-my-fix --patchBranch i111-my-fix-mutation --config config.xml --listOfProjects projects-to-test-on.properties
 ```
 
+## Javadoc regression:
+Many javadoc comments contain errors. To avoid report pollution with
+parsing errors, add suppression to the configuration:
+```
+<module name = "Checker">
+  <module name="SuppressionSingleFilter">
```

Comment:
Why checker filter over treewalker filter?

Reply:
better form of config is:
```
/var/tmp$ cat Test.java 
/** <a> bad html */
class Test {}

/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="SummaryJavadoc" />

    <module name="SuppressionXpathSingleFilter">
      <property name="message" value="Javadoc comment at column \d+ has parse error"/>
    </module>
  </module>
</module>

/var/tmp$ java -jar checkstyle-8.23-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```

We should use Trwwwalker filter, as we clearly know from where we want to suppress violations.

Reply:
One more step to improve config:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="haltOnException" value="false"/>
  <module name="TreeWalker">
    <module name="SummaryJavadoc" />

    <module name="SuppressionXpathSingleFilter">
      <property name="message" value="Javadoc comment at column \d+ has parse error"/>
    </module>
  </module>
  <module name="SuppressionSingleFilter">
    <property name="message" value="Exception occurred while parsing"/>
  </module>
</module>
```

this will suppress failures like - https://pbludov.github.io/issue-6929/apache-ant/index.html#A2
that damage diff report too. I propose to place it in README and reuse.

@rnveach , please review. It was long discussion on this for a while with you on this, with no agreement. Lets try to discuss this one more time.

Reply:
I am fine with new config.
The only possible addition I could see is ensuring the filter matches the "rule" on the report so we don't accidentally suppress something else by mistake. I believe it is the source, not sure if that matches `checks` in the filter. https://checkstyle.sourceforge.io/config_filters.html#SuppressionSingleFilter_Properties

Reply:
good point,
I added `<property name="checks" value="Checker"/>` to it, it works, tested on local.

Reply:
all done.
All changes applied, the section moved  to right after the command line switches section.